libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -126,13 +118,17 @@ HttpUrl generateRequestUrl() {
       .build();
   }
 
-  private void saveBlackList(List<String> revokedKeys) {
-    String fileContents = createListContent(revokedKeys);
+  private void saveBlackList(Response response) throws IOException {
+    Gson gson = new GsonBuilder().create();
+    JsonObject responseJson = gson.fromJson(response.body().string(), JsonObject.class);
+
+    responseJson = addSHAtoKeys(responseJson);
     FileOutputStream outputStream = null;
+    saveTimestamp();
 
     try {
       outputStream = context.openFileOutput(BLACKLIST_FILE, Context.MODE_PRIVATE);
-      outputStream.write(fileContents.getBytes());
+      outputStream.write(responseJson.toString().getBytes());
```

Comment:
just wondering whether we should encrypt this file?

Reply:
What would the security concern be over anyone knowing blacklisted keys? We already have all our keys viewable within the code itself. We also save this as `MODE_PRIVATE` which keeps it secure to the app. I believe this was acceptable to the security team during development of this feature.